spec/models/metric_spec.rb

```diff
@@ -1107,6 +1107,7 @@
         @vms_in_az = []
         2.times { @vms_in_az << FactoryGirl.create(:vm_openstack, :ems_id => @ems_openstack.id) }
         @availability_zone.vms = @vms_in_az
+        @availability_zone.vms.push(FactoryGirl.create(:vm_openstack, :ems_id => nil))
```

Comment:
the current issue is a vm gets through that has no ems.
I don't have access to an actual system that has this problem.

Maybe I misread the problem and it is somewhere else.
especially since @blomquisg pointed out we are traversing cloud_ems - so it has to have an ems